zebra-state/src/lib.rs

```diff
@@ -39,7 +39,9 @@ impl Config {
 impl Default for Config {
     fn default() -> Self {
         Self {
-            path: PathBuf::from("./.zebra-state"),
+            path: dirs::cache_dir()
+                .map(|dir| dir.join(".zebra").join("state"))
+                .unwrap_or_else(|| PathBuf::from(".zebra-state/")),
```

Comment:
Comment:

I think using the current directory for the state is ok for now. But using the current directory for a cache is non-standard, so it could surprise users. And it's more surprising if we only do it when we can't find the cache dir.

So once we add the state config to zebrad.toml (see #699's e8d6dfc0b2df5a2eb69091398c6d0b71ca034269), we should panic, and ask the user to configure an absolute directory.

Reply:
This is a good point, I think it should be fine to just switch to a panicking version immediately rather than wait for #699. Also I'm going to just go ahead and fix the issue I'm worried about with tests by adding an alternative method for setting up the cache dir via an env variable, that way I can make the test fns that @oxarbitrage is working on just set that env variable to the temp dir in the fn that constructs the `std::process::Command` objects for our tests.